util/stream.go

```diff
@@ -156,40 +136,76 @@ func (m *MessageStream) inbound() {
 			return
 		}
 
-		for i := 0; i < n; i++ {
-			if hdr < 4 {
-				hdrBuf[hdr] = tmpBuf[i]
-				buf.WriteByte(tmpBuf[i])
-				hdr += 1
-				if hdr >= 4 {
-					// MessageStream is not protocol agnostic. Reading length based
-					// on OpenFlow header field.
-					msgLen = int(binary.BigEndian.Uint16(hdrBuf[2:])) - 4
+		// Append the bytes read from the connection to buf.
+		buf.Write(tmpBuf[:n])
+
+		// Read from the connection until the OpenFlow message header is retrieved.
+		for buf.Len() >= 4 {
+			if totalLen == 0 {
```

Comment:
once the function is refactored a bit (https://github.com/antrea-io/libOpenflow/pull/80/files#r2324121263), we can probably look into moving some of that logic (for length computation) as a separate function for readability